src/test/java/io/debezium/connector/yugabytedb/YugabyteDBStreamConsistencyTest.java

```diff
@@ -180,4 +184,130 @@ public void recordsShouldStreamInConsistentOrderOnly() throws Exception {
+        int departmentId = 1;
+        int employeeId = 1, employeeBatchSize = 5 * scaleFactor;
+        int contractId = 1, contractBatchSize = 10 * scaleFactor;
+        int addressId = 1, addressBatchSize = 20 * scaleFactor;
+
+        // This counter will also indicate the final index of the inserted record while streaming.
+        long totalCount = 0;
+
+        // Lists to store the expected indices of the elements of respective tables in the final
+        // list of messages we will be receiving after streaming.
+        List<Integer> departmentIndices = new ArrayList<>();
+        List<Integer> employeeIndices = new ArrayList<>();
+        List<Integer> contractIndices = new ArrayList<>();
+        List<Integer> addressIndices = new ArrayList<>();
+
+        for (int i = 0; i < iterations; ++i) {
+            TestHelper.execute(String.format("INSERT INTO department VALUES (%d, 'my department no %d');", departmentId, departmentId));
+
+            // Inserting the index of the record for department table at its appropriate position.
+            departmentIndices.add((int) totalCount);
+            ++totalCount;
+
+            for (int j = employeeId; j <= employeeId + employeeBatchSize - 1; ++j) {
+                TestHelper.execute(String.format("INSERT INTO employee VALUES (%d, 'emp no %d', %d);", j, j, departmentId));
+                employeeIndices.add((int) totalCount);
+                ++totalCount;
+                for (int k = contractId; k <= contractId + contractBatchSize - 1; ++k) {
+                    TestHelper.execute(String.format("INSERT INTO contract VALUES (%d, 'contract no %d', %d);", k, k, j /* employee fKey */));
+                    contractIndices.add((int) totalCount);
+                    ++totalCount;
+
+                    for (int l = addressId; l <= addressId + addressBatchSize - 1; ++l) {
+                        TestHelper.execute(String.format("INSERT INTO address VALUES (%d, 'address no %d', %d);", l, l, k /* contract fKey */));
+                        addressIndices.add((int) totalCount);
+                        ++totalCount;
+                    }
+                    // Increment addressId for next iteration.
+                    addressId += addressBatchSize;
+                }
+                // Increment contractId for next iteration.
+                contractId += contractBatchSize;
+            }
+
+            // Increment employeeId for the next iteration
+            employeeId += employeeBatchSize;
+
+            // Increment department ID for more iterations
+            ++departmentId;
+        }
+
+        // Dummy wait
+        TestHelper.waitFor(Duration.ofSeconds(25));
+
+        List<SourceRecord> recordsToAssert = new ArrayList<>();
+
+        final long total = totalCount;
+        AtomicLong totalConsumedRecords = new AtomicLong();
+        try {
+            Awaitility.await()
+                    .atMost(Duration.ofSeconds(600))
```

Comment:
This is what we refactored our iterative functions to, if it remains a function of iteration then it will forever keep waiting for records if it gets stuck. We could have added another timeout in that iteration but then using Awaitility seemed a better idea - all the other tests currently follow the same.